tools/src/test/scala/com/nvidia/spark/rapids/tool/qualification/AppFilterSuite.scala

```diff
@@ -339,4 +369,64 @@ class AppFilterSuite extends FunSuite {
       }
     }
   }
+
+  test("standalone 1-oldest-per-app-name") {
+    val expected = Array("ndshours18", "ndsweeks2", "nds86")
+    testAppStartTimeAndPerAppName(appsFullWithFsToTest, "1-oldest-per-app-name", 3, expected)
+  }
+
+  test("standalone 2-newest-per-app-name") {
+    val expected = Array("ndshours18", "ndsweeks2", "nds86", "nds86")
+    testAppStartTimeAndPerAppName(appsFullWithFsToTest, "2-newest-per-app-name", 4, expected)
+  }
+
+  test("standalone 2-newest based on app time") {
+    val expected = Array("ndshours18", "nds86")
+    testAppStartTimeAndPerAppName(appsFullWithFsToTest, "2-newest", 2, expected)
+  }
+
+  test("standalone 3-oldest based on app time") {
+    val expected = Array("nds86", "nds86", "ndsweeks2")
+    testAppStartTimeAndPerAppName(appsFullWithFsToTest, "3-oldest", 3, expected)
+  }
+
+  test("standalone 10-oldest based on app time") {
+    val expected = Array("ndshours18", "nds86", "nds86", "ndsweeks2")
+    testAppStartTimeAndPerAppName(appsFullWithFsToTest, "10-oldest", 4, expected)
+  }
+
+  private def testAppStartTimeAndPerAppName(
```

Comment:
this isn't testing app start time (--start-app-time) lets update name

Reply:
Updated it. Not sure if it is still confusing :D